code/model/modelanimation_segments.cpp

```diff
@@ -53,93 +53,203 @@ namespace animation {
 		for (const auto& segment : m_segments) {
 			float timeLocal = time;
 			//Make sure that no segment runs over its length
 			if (timeLocal > segment->getDuration(pmi_id))
 				timeLocal = segment->getDuration(pmi_id);
 			
-			ModelAnimationData<true> deltaLocal = segment->calculateAnimation(base, timeLocal, pmi_id);
-			delta.applyDelta(deltaLocal);
+			segment->calculateAnimation(base, timeLocal, pmi_id);
 		}
-
-		return delta;
 	}
 
-	void ModelAnimationSegmentParallel::executeAnimation(const ModelAnimationData<>& state, float timeboundLower, float timeboundUpper, ModelAnimationDirection direction, int pmi_id) {
+	void ModelAnimationSegmentParallel::executeAnimation(const ModelAnimationSubmodelBuffer& state, float timeboundLower, float timeboundUpper, ModelAnimationDirection direction, int pmi_id) {
 		for (const auto& segment : m_segments) {
 			if (timeboundLower < segment->getDuration(pmi_id)) {
 				segment->executeAnimation(state, timeboundLower, fminf(timeboundUpper, segment->getDuration(pmi_id)), direction, pmi_id);
 			}
 		}
 	}
 
+	void ModelAnimationSegmentParallel::exchangeSubmodelPointers(ModelAnimationSet& replaceWith) {
+		for (const auto& segment : m_segments)
+			segment->exchangeSubmodelPointers(replaceWith);
+	}
+
 	void ModelAnimationSegmentParallel::addSegment(std::shared_ptr<ModelAnimationSegment> segment) {
 		m_segments.push_back(std::move(segment));
 	}
 
+	ModelAnimationParseHelper::Registrar ModelAnimationSegmentParallel::reg("$Segment Parallel:", &parser);
+	std::shared_ptr<ModelAnimationSegment> ModelAnimationSegmentParallel::parser(ModelAnimationParseHelper* data) {
+		auto submodelOverride = ModelAnimationParseHelper::parseSubmodel();
+
+		ignore_white_space();
+		auto segment = std::shared_ptr<ModelAnimationSegmentParallel>(new ModelAnimationSegmentParallel());
+
+		while (!optional_string("+End Segment")) {
+			if (submodelOverride)
+				data->parentSubmodel = submodelOverride;
+
+			segment->addSegment(data->parseSegment());
+			ignore_white_space();
+		}
+
+		return segment;
+	}
+
 
 	ModelAnimationSegmentWait::ModelAnimationSegmentWait(float time) : m_time(time) { }
 
-	void ModelAnimationSegmentWait::recalculate(const submodel_instance* /*submodel_instance*/, const bsp_info* /*submodel*/, const ModelAnimationData<>& /*base*/, int pmi_id) { 
-		m_duration[pmi_id] = m_time;
+	ModelAnimationSegment* ModelAnimationSegmentWait::copy() const {
+		return new ModelAnimationSegmentWait(*this);
+	}
+	
+	void ModelAnimationSegmentWait::recalculate(ModelAnimationSubmodelBuffer& /*base*/, polymodel_instance* pmi) {
+		m_duration[pmi->id] = m_time;
 	};
 
+	ModelAnimationParseHelper::Registrar ModelAnimationSegmentWait::reg("$Wait:", &parser);
+	std::shared_ptr<ModelAnimationSegment> ModelAnimationSegmentWait::parser(ModelAnimationParseHelper* /*data*/) {
+		required_string("+Time:");
+		float time = 0.0f;
+		stuff_float(&time);
```

Comment:
Since the whole animation system currently uses floating point time, it's probably not currently useful.
_Should_ precision become an issue here, which I doubt, then it might be wise to retool this to fixed-point values.